mne/encoding/feature.py

```diff
@@ -0,0 +1,298 @@
+#          Jona Sassenhagen <jona.sassenhagen@gmail.de>
+#          Jean-Remi King <jeanremi.king@gmail.com>
+#
+# License: BSD (3-clause)
+import numpy as np
+from scipy import sparse
+from ..utils import warn
+
+
+class FeatureDelayer(object):
+    """Creates delayed versions of an input stimulus.
+
+    Parameters
+    ----------
+    delays: array of floats, shape (n_delays,)
+        The time (in seconds) of each delay for specifying
+        pre-defined delays. Negative means timepoints in the past,
+        positive means timepoints in the future.
+    sfreq: float
+        The sampling frequency of the series. Defaults to 1.0
+    """
+    def __init__(self, delays=None, sfreq=1.):
+        delays, sfreq = _check_delayer_params(
+            delays, sfreq)
+        self.delays = delays
+        self.sfreq = float(sfreq)
+
+    def fit(self, X, y=None):
+        """Does nothing, only used for sklearn compatibility.
+
+        Parameters
+        ----------
+        X : array
+        y : array
+
+        Returns
+        -------
+        self
+        """
+        return self
+
+    def transform(self, X=None, y=None):
+        """Create a time-lagged representation of X.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_times, n_features)
+            The input data to be time-lagged and returned.
+
+        Returns
+        -------
+        X_delayed : array, shape (n_times, n_feature * n_delays)
+            The delayed featuers of X.
+        """
+        X_delayed, names = delay_timeseries(X, delays=self.delays,
+                                            sfreq=self.sfreq)
+        self.delays_ = names
+        return X_delayed
+
+    def fit_transform(self, X, y=None):
+        """Does nothing, only used for sklearn compatibility.
+
+        Parameters
+        ----------
+        X : array
+        y : array
+
+        Returns
+        -------
+        self
+        """
+        return self.transform(X)
+
+
+class EventsBinarizer(object):
+    """Create a continuous-representation of event onsets.
+
+    Parameters
+    ----------
+    n_times : int
+        The total number of times in the output array.
+    sfreq : float
+        The sampling frequency used to convert times to sample indices.
+        Defaults to 1.0
+    sparse : bool
+        Whether to output continuous events as a dense array or a sparse
+        matrix. Defaults to False.
+    """
+    def __init__(self, n_times, sfreq=1., sparse=False):
+        self.n_times = n_times
+        self.sfreq = sfreq
+        self.feature_type = 'event_id'
+        self.sparse = sparse
+
+    def fit(self, X=None, y=None):
+        """Does nothing, only used for sklearn compatibility.
+
+        Parameters
+        ----------
+        X : array
+        y : array
+
+        Returns
+        -------
+        self
+        """
+        return self
+
+    def transform(self, event_ixs, event_ids=None, event_dict=None,
```

Comment:
so only `transform(x)` with `x` being a 1-D vector of event times? No support for multiple event types?